SwanCustomEnvironments/swancustomenvironments/scripts/builders/accpy.sh

```diff
@@ -1,24 +1,31 @@
 #!/bin/bash
 
+# Create a middle layer for installing Spark extensions, putting them apart from the user environment
+uv venv $SWAN_ENV --seed 2>&1
+source $SWAN_ENV/bin/activate
+uv pip install "ipykernel==${IPYKERNEL_VERSION}"
+SWAN_PACKAGES_PATH=$(python3 -c 'import sysconfig; print(sysconfig.get_paths()["purelib"])')
+deactivate
+
 # If using NXCALS, we need to install the Spark extensions and the nxcals package.
 if [ -n "${INSTALL_NXCALS}" ]; then
+    # Check if the requirements file contains the nxcals package
```

Comment:
Instead of grepping here, can we detect that nxcals is not there when we parse the spark config? If spark-default is not there, we complain about nxcals absence. I don't recall now when that is done.

Reply:
That is done later, after the env gets created. But we can do something better indeed.